thriftbp/client_middlewares.go

```diff
@@ -398,3 +400,25 @@ func PrometheusClientMiddleware(remoteServerSlug string) thrift.ClientMiddleware
 		}
 	}
 }
+
+func getClientError(result thrift.TStruct, err error) error {
+	if err != nil {
+		return err
+	}
+	v := reflect.Indirect(reflect.ValueOf(result))
+	if v.Kind() != reflect.Struct {
+		return nil
+	}
+	typ := v.Type()
+	for i := 0; i < v.NumField(); i++ {
+		if typ.Field(i).Name == "Success" {
```

Comment:
Is this just an optimization? I feel like line 419 will also skip all of the success fields since they won't have any methods.

Reply:
this is more of "in theory people can define an exception type as their rpc return type and we could accidentally use that as error" (e.g. currently in baseplate.thrift's `is_healthy` endpoint, the return type is `bool`, but in theory someone can define an endpoint with a return type being also an exception)